qcodes_contrib_drivers/drivers/Oxford/Triton.py

```diff
@@ -0,0 +1,213 @@
+
+        self.add_parameter(name='pressure_mixture_tank',
+                           unit='Bar',
+                           get_parser=float,
+                           get_cmd=lambda: self.get_pressure('tank'),
+                           docstring='Pressure of the mixture tank',)
+
+        self.add_parameter(name='pressure_forepump_back',
+                           unit='Bar',
+                           get_parser=float,
+                           get_cmd=lambda: self.get_pressure('forepump'),
+                           docstring='Pressure of the forepump back',)
+
+        self.add_parameter(name='temperature_50k_plate',
+                           unit='K',
+                           get_parser=float,
+                           get_cmd=lambda: self.get_temperature('50k'),
+                           docstring='Temperature of the 50K plate',)
+
+        self.add_parameter(name='temperature_4k_plate',
+                           unit='K',
+                           get_parser=float,
+                           get_cmd=lambda: self.get_temperature('4k'),
+                           docstring='Temperature of the 4K plate',)
+
+        if magnet:
+            self.add_parameter(name='temperature_magnet',
+                               unit='K',
+                               get_parser=float,
+                               get_cmd=lambda: self.get_temperature('magnet'),
+                               docstring='Temperature of the magnet',)
+
+        self.add_parameter(name='temperature_still',
+                           unit='K',
+                           get_parser=float,
+                           get_cmd=lambda: self.get_temperature('still'),
+                           docstring='Temperature of the still',)
+
+        self.add_parameter(name='temperature_100mk',
+                           unit='K',
+                           get_parser=float,
+                           get_cmd=lambda: self.get_temperature('100mk'),
+                           docstring='Temperature of the 100mk plate',)
+
+        self.add_parameter(name='temperature_mixing_chamber',
+                           unit='K',
+                           get_parser=float,
+                           get_cmd=lambda: self.get_temperature('mc'),
+                           docstring='Temperature of the mixing chamber',)
+        
+        self.connect_message()
+
+    def vcl2csv(self) -> Optional[str]:
+        """
+        Convert vcl file into csv file using proprietary binary exe.
+        The executable is called through the python subprocess library.
+        To avoid to frequent file conversion, a timer of self.conversion_timer
+        second is used.
+
+        Returns:
+            str: The output of the bash command
+        """
+        
+        conversion = False
+        if self._timer+self.conversion_timer <= time.time():
+            conversion = True
+        elif not os.path.isfile(self.file_path[:-3]+'txt'):
```

Comment:
The expression will never be true as it has already been checked if given file_path is correct in line 54.

Reply:
The test line 35 tests if a vcl file path is correctly given.
That vcl file is the original oxford log file.

The test line 135 tests if a txt file with the same file path as the vcl file is present.
That txt file is created by the driver through the binary vcl converter.
That condition is true the first time the function is called, when the timer hasn't reached is threshold but no txt file already exist.

Does that make sense?

Reply:
Sorry, I should have been more clear. I meant expression in line 136 `not os.path.isfile(self.file_path[:-3]+'txt')`. You have already checked for it in line 54.

Reply:
`file_path` is the path towards the **vcl** file.

`not os.path.isfile(self.file_path[:-3]+'txt')` check the presence of the converted **txt** file.